File: files/CSEC 12.05.py
import pandas as pd

input = pd.read_excel("files/Excel Challenge 12th May.xlsx", sheet_name="Sheet1", skiprows=2, usecols="B:E")

result = input.melt(id_vars=["Doctors", "Session"], var_name="Day", value_name="Patient")
result["Appointment"] = result["Doctors"] + " - " + result["Session"] + " - " + result["Day"]
result = result.sort_values(by=["Patient", "Appointment"])[["Patient", "Appointment"]].reset_index(drop=True)

print(result)

File: files/2024-05-12/CSEC 12.05.R
library(tidyverse)
library(readxl)

input = read_excel("files/Excel Challenge 12th May.xlsx", range = "B3:E13")

result = input %>%
  pivot_longer(cols = -c("Doctors", "Session"), names_to = "Day", values_to = "Patient") %>%
  unite("Appointment", c("Doctors", "Session", "Day"), sep = " - ") %>%
  arrange(Patient, Appointment) %>%
  select(Patient, Appointment)
